tools/ExtensionLibrary/DevHome.ExtensionLibrary/ViewModels/ExtensionLibraryViewModel.cs

```diff
@@ -52,10 +58,60 @@ private async void OnPluginsChanged(object? sender, EventArgs e)
         await _dispatcher.EnqueueAsync(() =>
         {
             ShouldShowStoreError = false;
+            GetInstalledExtensions();
             GetAvailablePackages();
         });
     }
 
+    private void GetInstalledExtensions()
+    {
+        var extensionWrappers = Task.Run(async () =>
+        {
+            var pluginService = Application.Current.GetService<IPluginService>();
+            return await pluginService.GetInstalledPluginsAsync(true);
+        }).Result;
+
+        InstalledPackagesList.Clear();
+
+        foreach (var extensionWrapper in extensionWrappers)
+        {
+            // Don't show self as an extension.
+            if (Package.Current.Id.FullName == extensionWrapper.PackageFullName)
+            {
+                continue;
+            }
+
+            var hasSettingsProvider = extensionWrapper.HasProviderType(Microsoft.Windows.DevHome.SDK.ProviderType.Settings);
+            var extension = new InstalledExtensionViewModel(extensionWrapper.Name, extensionWrapper.PackageFullName, hasSettingsProvider);
+
+            // Each extension is shown under the package that contains it. Search to see if we have the package in the
+            // list already and add the extension to that package in the list if we do.
+            var foundPackage = false;
+            foreach (var installedPackage in InstalledPackagesList)

```

Comment:
nit:
```cs
var package = InstalledPackagesList.FirstOrDefault(p => p.PackageFamilyName == extensionWrapper.PackageFamilyName);
if(package == null)
{
    // Create package
}

package.InstalledExtensionsList.Add(extension);
```

Reply:
Agree with this, also agree with using package family instead of package full name. Should be functionally equivalent but generally recommend avoiding using full name unless version matters and you really need it. Version doesn't matter here, we just want to map up the extension with whatever the installed version is.